### PR TITLE
补充js和ts对应的哈希查找

### PR DESCRIPTION
**Pull request by zhuoqinyue**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/typescript/chapter_searching/hashing_search.ts
Comment (krahets) [inline codes/typescript/chapter_searching/hashing_search.ts:35]: Uncomment them

Comment (zhuoqinyue) [reply codes/typescript/chapter_searching/hashing_search.ts:35]: Maybe i will delete them.
There no exists the arrToLinkedList method in `typescript\module\ListNode.ts`.
I could translate this method in next week.

Comment (krahets) [reply codes/typescript/chapter_searching/hashing_search.ts:35]: OK. Feel free to add codes to this PR.

**Comment by zhuoqinyue** [comment]
Hey, I've translated this method.

File: codes/javascript/chapter_searching/hashing_search.js
Comment (krahets) [inline codes/javascript/chapter_searching/hashing_search.js:1]: /**

File: codes/javascript/chapter_searching/hashing_search.js
Comment (krahets) [inline codes/javascript/chapter_searching/hashing_search.js:13]: Four spaces for the code indentation.

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

Wait for the final review by @justin-tse .

**Comment by zhuoqinyue** [comment]
OK. I've fixed all.
Then I have a suggestion that you can clarify the code style in `chapter_preface` as far as possible.

**Comment by krahets** [comment]
> OK. I've fixed all. Then I have a suggestion that you can clarify the code style in `chapter_preface` as far as possible.

Got it.

File: codes/typescript/chapter_searching/hashing_search.ts
Comment (justin-tse) [inline codes/typescript/chapter_searching/hashing_search.ts:15]:
```suggestion
function hashingSearch(map: Map<number, number>, target: number): number {
    // 哈希表的 key: 目标元素，value: 索引
    // 若哈希表中无此 key ，返回 -1
    return map.has(target) ? map.get(target) as number : -1;
```

File: codes/typescript/chapter_searching/hashing_search.ts
Comment (justin-tse) [inline codes/typescript/chapter_searching/hashing_search.ts:22]:
```suggestion
function hashingSearch1(map: Map<number, ListNode>, target: number): ListNode | null {
    // 哈希表的 key: 目标结点值，value: 结点对象
    // 若哈希表中无此 key ，返回 null
    return map.has(target) ? map.get(target) as ListNode : null;
```

File: docs/chapter_searching/hashing_search.md
Comment (justin-tse) [inline docs/chapter_searching/hashing_search.md:86]:
```suggestion
    function hashingSearch(map: Map<number, number>, target: number): number {
        // 哈希表的 key: 目标元素，value: 索引
        // 若哈希表中无此 key ，返回 -1
        return map.has(target) ? map.get(target) as number : -1;
```

File: docs/chapter_searching/hashing_search.md
Comment (justin-tse) [inline docs/chapter_searching/hashing_search.md:179]:
```suggestion
    function hashingSearch1(map: Map<number, ListNode>, target: number): ListNode | null {
        // 哈希表的 key: 目标结点值，value: 结点对象
        // 若哈希表中无此 key ，返回 null
        return map.has(target) ? map.get(target) as ListNode : null;
```

**Review by justin-tse — APPROVED** [review]
Thank you @zhuoqinyue for your clear code. It looks good and only has a small flaw.

**Comment by zhuoqinyue** [comment]
Hey, I've fixed all.
And should I do something in PR #121 and #120? @justin-tse 
I will translate the `Deque` after this PR finish

**Comment by justin-tse** [comment]
@zhuoqinyueThank you!  Of course! I will assign the task of Deque to you.

**Review by zhuoqinyue — COMMENTED** [review]
I've fixed all

**Review by krahets — APPROVED** [review]

**Comment by krahets** [comment]
> I have a question. How could I close this? image @krahets

Hi! You shouldn't have to. After approval by @justin-tse, the PR will be merged soon.

**Pull request merged**
